src/pages/user-descriptor/line-graph.jsx

```diff
@@ -1,59 +1,45 @@
 import React from "react";
 import ReactEcharts from "echarts-for-react";
 
-// Expected Object
-function BossObject(Unit, Weight, Latitude, Longitude, Time, Gender) {
-  this.unit = Unit;
-  this.weight = Weight;
-  this.latitude = Latitude;
-  this.longitude = Longitude;
-  this.time = Time;
-  this.gender = Gender;
+// GraphQL
+import { gql } from "apollo-boost";
+import { useQuery } from "@apollo/react-hooks";
+
+const lgData = gql`
+  {
+    getPaginatedDescriptors(unit: "lb", start: 0, count: 400) {
+      value
+      unixTimestamp
+    }
+  }
+`;
+
+function DataFunction() {
+  const { loading, error, data } = useQuery(lgData);
+
+  if (loading) return "Loading...";
+  if (error) return `Error! ${error.message}`;
```

Comment:
This function is a huge problem. The reason is that you are returning an array on success, and strings on loading and error. Other parts of your code that uses this has the assumption that this ONLY returns an array. If you use any of the operations that you can do on an array on a string, the entire application will crash. You need to make it so that your code adjusts the view (your line graph) based on what is returned. If loading or error is returned, show that. If data is returned show that. If you need to mock that scenario, change the mock server temporarily so that it returns an error or add some sort of timeout delay so that you can see it loading.